src/librbd/api/Mirror.cc

```diff
@@ -434,10 +434,30 @@ int Mirror<I>::image_enable(I *ictx, mirror_image_mode_t mode,
     return r;
   }
 
-  if (mirror_mode != cls::rbd::MIRROR_MODE_IMAGE) {
-    lderr(cct) << "cannot enable mirroring in the current pool mirroring mode"
-               << dendl;
-    return -EINVAL;
+  if (ictx->md_ctx.get_namespace().empty()) {
+    if (mirror_mode == cls::rbd::MIRROR_MODE_DISABLED) {
+      lderr(cct) << "cannot enable mirroring: mirroring is not enabled on a "
+                    "pool" << dendl;
+      return -EINVAL;
+    }
+
+    if (mirror_mode != cls::rbd::MIRROR_MODE_IMAGE) {
+      lderr(cct) << "cannot enable mirroring: pool is not in image mirror mode"
+                 << dendl;
+      return -EINVAL;
+    }
+  } else {
```

Comment:
Let's avoid repetition by introducing `pool_or_namespace` helper that would return either `"pool"` or `"namespace"` and use it here as follows:
```
      lderr(cct) << "cannot enable mirroring: mirroring is not enabled on a "
                 << pool_or_namespace(ictx) << dendl;

```